compiler/rustc_mir/src/transform/inline.rs

```diff
@@ -492,6 +493,34 @@ impl Inliner<'tcx> {
                     }
                 }
 
+                // If there are any locals without storage markers, give them storage only for the
+                // duration of the call.
+                for local in callee_body.vars_and_temps_iter() {
```

Comment:
Locals without storage markers are often on the unwind path where they are deliberately not emitted.